packages/datagateway-dataview/src/views/datafilePreview/previewDatafileButton.component.tsx

```diff
@@ -0,0 +1,50 @@
+import VisibilityIcon from '@mui/icons-material/Visibility';
+import { Box, IconButton, Tooltip } from '@mui/material';
+import type { Datafile } from 'datagateway-common';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { Link } from 'react-router-dom';
+import { isDatafilePreviewable } from './datafileExtension';
+
+export interface PreviewDatafileButtonProps {
+  datafile: Datafile;
+}
+
+/**
+ * A button that shows the preview of the given {@link Datafile} when clicked.
+ */
+function PreviewDatafileButton({
+  datafile,
+}: PreviewDatafileButtonProps): JSX.Element {
+  const [t] = useTranslation();
+
+  const isSupported = isDatafilePreviewable(datafile);
+
+  return (
+    <Link
+      to={`datafile/${datafile.id}`}
```

Comment:
Clicking on the preview button for some reason causes the whole page to reload for me, despite the fact that this is using a `react-router` link. If you instead use the `component` prop of `IconButton` to render that as a `Link`, I think it should work fine